### PR TITLE
[risk=low][RW-5616] Add institution category to Google Analytics

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Following the pattern of #2699, add a Google Analytics Custom Dimension and associated `analytics.ts` function to determine the Institutional Category for the logged-in user:
* **Operations** if the user has a Verified institutional Affiliation with the magic-short-name 'AouOps' institution
* **Researcher** if the user has a Verified institutional Affiliation with any other institution
* **No Institution** if the user has none
* **Unknown** for any GA events triggered before user sign-in

Tested by observing all 4 states on Local using [Google Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna)

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [x] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/utils/analytics.ts [hunk 1]
```diff
@@ -158,21 +159,59 @@ enum UserAuthState {
...
+    [environment.gaLoggedInDimension]: UserAuthState.LOGGED_OUT,
+    [environment.gaUserInstitutionCategoryDimension]: InstitutionCategoryState.UNKNOWN,
   });
   gtag('config', environment.gaId);
 }
+
+// invokes the 'set' command in gtag.js which mutates global state.
+// All gtag event triggers after this call will be affected.
+// see https://developers.google.com/gtagjs/reference/api
+function gtagSet(setParam: object) {
```
Comment (jmthibault79) [inline ui/src/app/utils/analytics.ts:215]: The behavior of this was surprising enough to me to be worth calling out and centralizing

File: ui/src/environments/environment.perf.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export const environment: Environment = {
   gaId: 'UA-112406425-2',
   gaUserAgentDimension: 'dimension1',
   gaLoggedInDimension: 'dimension2',
+  gaUserInstitutionCategoryDimension: 'dimension3',   // TODO create
```
Comment (jmthibault79) [inline ui/src/environments/environment.perf.ts:16]: I haven't created these yet in GA.  Reminder: I need to do that and update this text

**Review by gjuggler — APPROVED** [review]
Looks good!

**Comment by jmthibault79** [comment]
The e2e error is a Rawls timeout.  No connection to this PR.

**Pull request merged**